spring-data-mongodb/src/main/java/org/springframework/data/mongodb/core/mapping/MongoMappingContext.java

```diff
@@ -145,4 +150,10 @@ public void setAutoIndexCreation(boolean autoCreateIndexes) {
 
 		return new UnwrappedMongoPersistentEntity<>(entity, new UnwrapEntityContext(persistentProperty));
 	}
+
+	@Nullable
+	public ApplicationContext getApplicationContext() {
```

Comment:
This doesn't seem seem to be necessary as we do not have any callers to `getApplicationContext()`